src/contexts/makeDispatchWrapper.js

```diff
@@ -0,0 +1,206 @@
+import axios from 'axios';
+import { isPresentInArray } from './arrayUtils';
+
+export function makeDispatchWrapper(state, dispatch) {
+	async function dispatchWrapper(action) {
```

Comment:
Its really nice that you wrapped your reusable code inside a function and took it in separate file. But i have a suggestion here, dispatchWrapper does not convey the actual action it will do after invoke, you can change this name so that it will be more descriptive.